dangerzone/updater/cli.py

```diff
@@ -0,0 +1,179 @@
+    signatures.store_signatures(sigs, manifest_digest, update_logindex=False)
+    click.echo(f"✅ Signatures have been verified and stored locally")
+
+
+@main.command()
+@click.argument("archive_filename", type=click.Path(exists=True))
+@click.option("--force", is_flag=True)
+def load_archive(archive_filename: Path, force: bool) -> None:
+    """Upgrade the local image to the one in the archive."""
+    try:
+        loaded_image = signatures.upgrade_container_image_airgapped(
+            archive_filename, bypass_logindex=force
+        )
+        click.echo(
+            f"✅ Installed image {archive_filename} on the system as {loaded_image}"
+        )
+    except errors.ImageAlreadyUpToDate as e:
+        click.echo(f"✅ {e}")
+    except errors.InvalidLogIndex as e:
+        click.echo("❌ Trying to install image older that the currently installed one")
+        raise click.Abort()
+    except errors.SignatureError as e:
+        click.echo(f"❌ Failed to verify the signatures.")
+        raise click.Abort()
+
+
+@main.command()
+@click.option("--image", default=DEFAULT_IMAGE_NAME)
+@click.option("--output", default="dangerzone-{arch}.tar")
+@click.option(
+    "--arch",
+    default=get_architecture(),
+    help="The architecture you want to prepare the archive for. By default, it uses your platform.",
+)
+def prepare_archive(image: str, output: str, arch: str) -> None:
+    """Prepare an archive to upgrade the dangerzone image (useful for airgapped environment)"""
+    archive = output.format(arch=arch)
+    try:
+        signatures.prepare_airgapped_archive(image, archive, arch)
+        click.echo(f"✅ Archive {archive} created")
+    except errors.SignatureError:
+        click.echo("❌ Failed to verify the signatures.")
+        raise click.Abort()
+
+
+@main.command()
+@click.argument("image", default=DEFAULT_IMAGE_NAME)
+def verify_local(image: str, pubkey: Path) -> None:
+    """
+    Verify the local image signature against a public key and the stored signatures.
+    """
+    if signatures.verify_local_image(image):
+        click.echo(
+            (
+                f"Verifying the local image:\n\n"
+                f"pubkey: {DEFAULT_PUBKEY_LOCATION}\n"
+                f"image: {image}\n\n"
+                f"✅ The local image {image} has been signed with the public key"
+            )
+        )
+
+
+@main.command()
+@click.argument("image_name")
+@click.option(
+    "--branch",
+    default=DEFAULT_BRANCH,
+    help="The Git branch that the image was built from",
+)
+@click.option(
+    "--commit",
+    required=True,
+    help="The Git commit the image was built from",
+)
+@click.option(
+    "--repository",
+    default=DEFAULT_REPOSITORY,
+    help="The github repository to check the attestation for",
+)
+@click.option(
+    "--workflow",
+    default=".github/workflows/release-container-image.yml",
+    help="The path of the GitHub actions workflow this image was created from",
+)
+def attest_provenance(
```

Comment:
It would be nice to test this CLI command, same way as we test `./dev_scripts/reproduce-image.py` in `build-push-image.yml`. Are you ok if I open an issue so that we can revisit later?

Reply:
Yep, good idea! Just did so!